internal/federationin/federationin.go

```diff
@@ -212,10 +215,73 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 type pullOptions struct {
-	deps           pullDependencies
-	query          *model.FederationInQuery
-	batchStart     time.Time
-	truncateWindow time.Duration
+	deps                         pullDependencies
+	query                        *model.FederationInQuery
+	batchStart                   time.Time
+	truncateWindow               time.Duration
+	maxIntervalStartAge          time.Duration
+	maxMagnitudeSymptomOnsetDays uint
+	debugReleaseSameDay          bool
+}
+
+func updateTimestamps(max, maxRevised time.Time, state *federation.FetchState) (time.Time, time.Time) {
+	if state == nil {
+		// Didn't get any response, don't advance timestamps.
+		return max, maxRevised
+	}
+
+	if state.KeyCursor != nil {
+		if timestamp := time.Unix(state.KeyCursor.Timestamp, 0).UTC(); timestamp.After(max) {
+			max = timestamp
+		}
+	}
+	if state.RevisedKeyCursor != nil {
+		if revisedTimestamp := time.Unix(state.RevisedKeyCursor.Timestamp, 0).UTC(); revisedTimestamp.After(maxRevised) {
+			maxRevised = revisedTimestamp
+		}
+	}
+	return max, maxRevised
+}
+
+// converts a federation ExposureKey to a model Exposure.
+func buildExposure(e *federation.ExposureKey) *publishmodel.Exposure {
+	upperRegions := make([]string, len(e.Regions))
+	for i, r := range e.Regions {
+		upperRegions[i] = strings.ToUpper(strings.TrimSpace(r))
+	}
+	sort.Strings(upperRegions)
+
+	exposure := publishmodel.Exposure{
+		ExposureKey:      e.ExposureKey,
+		TransmissionRisk: int(e.TransmissionRisk),
+		Regions:          upperRegions,
+		Traveler:         e.Traveler,
+		IntervalNumber:   e.IntervalNumber,
+		IntervalCount:    e.IntervalCount,
+		LocalProvenance:  false,
+	}
+	switch e.ReportType {
+	case federation.ExposureKey_CONFIRMED_TEST:
+		exposure.ReportType = verifyapi.ReportTypeConfirmed
+	case federation.ExposureKey_CONFIRMED_CLINICAL_DIAGNOSIS:
+		exposure.ReportType = verifyapi.ReportTypeClinical
+	case federation.ExposureKey_REVOKED:
+		exposure.ReportType = verifyapi.ReportTypeNegative
+	case federation.ExposureKey_SELF_REPORT:
```

Comment:
Does this need to be guarded with the configuration option? It seems like that config is unused if we're transforming self and recursive to clinical here.

Reply:
yep - I missed those.